### PR TITLE
Add optional param, payerEmail, to PayPalVaultRequest

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Add the optional `payer_email` to the PayPal vault request

### Checklist

 - ~[ ] Added a changelog entry~
 - ~[ ] Relevant test coverage~

### PayPal and Braintree engineering requirements
 > If you're an engineer for PayPal or Braintree, complete this section. For PRs from the community, please ignore!

**When do these changes need to be released?** 
[ ] ASAP (let's discuss outside this PR)
[ ] Soon, here's our target release date: <DD/MM/YYYY>
[ ] Whenever, no rush

### CODE REVIEW

File: PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java [hunk 1]
```diff
@@ -26,6 +28,16 @@ public void setShouldOfferCredit(boolean shouldOfferCredit) {
         this.shouldOfferCredit = shouldOfferCredit;
     }
 
+    /**
+     * Optional: User email to initiate a quicker authentication flow in cases where the user has a
+     * PayPal Account with the same email.
+     *
+     * @param payerEmail - email address of the payer
+     */
+    public void setPayerEmail(String payerEmail) {
```
[the remark this answers is not in the document]
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java:41]: Good catch, I completely forgot about the Kotlin interop. I just added the missing getter.

Would we still need to update nullability parameters?

[the remark this answers is not in the document]
Comment (sarahkoop) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java:41]: Hmm I can't recall off the top of my head if the nullability param is required, or just required not to be different on the getter vs setter. For consistency across the rest of the SDK maybe just add the nullable param to both?

[the remark this answers is not in the document]
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java:41]:
```kotlin
val request = PayPalVaultRequest()
request.payerEmail = ""
```

This seemed to work in Kotlin by adding the missing getter. Was there another reason why we made the setter param nullable?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Braintree Android SDK Release Notes
 
+## unreleased
+* Add optional property `PayPalVaultRequest.setPayerEmail()`
```
Comment (jaxdesmarais) [inline CHANGELOG.md:4]: nit to match the changelog formatting used previously:
```suggestion
## unreleased

* PayPal
  * Add optional property `PayPalVaultRequest.setPayerEmail()`
```

File: PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java [hunk 3]
```diff
@@ -14,6 +14,8 @@ public class PayPalVaultRequest extends PayPalRequest implements Parcelable {
 
     private boolean shouldOfferCredit;
 
+    private String payerEmail;
```
Comment (jaxdesmarais) [inline PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java:17]: We started a discussion on the iOS PR about naming, not sure if you have thoughts but we should align the name across platforms: https://github.com/braintree/braintree_ios/pull/1188#discussion_r1486758270

**Review by sarahkoop — APPROVED** [review]

File: PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java (hunk 3 above)
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalVaultRequest.java:17]: Thanks for calling this out! I've added a do not merge label, will make sure to align with iOS before this goes in.

**Review by agedd — APPROVED** [review]
🚀

**Review by jaxdesmarais — APPROVED** [review]

**Comment by scannillo** [comment]
Updating the base branch to target the ShopperInsights feature work. Passing this email will be part of the E2E testing teams verification before release.

**Pull request merged**